packages/components/src/validated-form-controls/components/stories/overview.story.tsx

```diff
@@ -437,3 +439,154 @@ export const ValidateOnPopoverClose: StoryObj< typeof ValidatedInputControl > =
 			},
 		},
 	};
+
+/**
+ * Sometimes, a field's validity may depend on the state of other fields.
+ *
+ * In this example, the text field must start with "http" if the checkbox is checked.
+ * Validator logic should be added to both fields.
+ */
+export const ValidateWithOtherFieldState: StoryObj<
+	typeof ValidatedInputControl
+> = {
+	decorators: formDecorator,
+	render: function Template( { ...args } ) {
+		const [ text, setText ] = useState( '' );
+		const [ isHttp, setIsHttp ] = useState( false );
+		const [ customValidity, setCustomValidity ] =
+			useState<
+				React.ComponentProps<
+					typeof ValidatedInputControl
+				>[ 'customValidity' ]
+			>( undefined );
+
+		return (
+			<>
+				<ValidatedInputControl
+					{ ...args }
+					value={ text }
+					onChange={ ( newValue ) => {
+						setText( newValue ?? '' );
+					} }
+					onValidate={ ( value ) => {
+						if ( isHttp && value && ! value.startsWith( 'http' ) ) {
+							setCustomValidity( {
+								type: 'invalid',
+								message: 'Text must start with "http".',
+							} );
+						} else {
+							setCustomValidity( undefined );
+						}
+					} }
+					customValidity={ customValidity }
+				/>
+				<CheckboxControl
+					__nextHasNoMarginBottom
+					label="Text starts with HTTP"
+					checked={ isHttp }
+					onChange={ ( value ) => {
+						setIsHttp( value );
+
+						if (
+							value === true &&
+							text &&
+							! text.startsWith( 'http' )
+						) {
+							setCustomValidity( {
+								type: 'invalid',
+								message: 'Text must start with "http".',
+							} );
+						} else {
+							setCustomValidity( undefined );
+						}
+					} }
+				/>
+			</>
+		);
+	},
+	args: {
+		label: 'Text',
+	},
+};
+
+/**
+ * When a field's validity can only be determined after the submit button is clicked,
+ * `customValidity` can be set within the form's `onSubmit` handler.
+ *
+ * In this example, the text field must start with "http" if the submit button is clicked,
+ * but can be any string if the "Create new page" button is clicked.
+ */
+export const ValidateOnSubmit: StoryObj< typeof ValidatedInputControl > = {
+	render: function Template( { ...args } ) {
+		const [ text, setText ] = useState( '' );
+		const [ customValidity, setCustomValidity ] =
+			useState<
+				React.ComponentProps<
+					typeof ValidatedInputControl
+				>[ 'customValidity' ]
+			>( undefined );
+
+		return (
+			<form
```

Comment:
Is presence of `form` a prerequisite or could this theoretically work without a form? So for example the button or `ENTER` are responsible for submitting.

Reply:
I do generally recommended using forms, as there are a lot of built-in behaviors you can leverage (not to mention supporting "`ENTER` to submit" automatically). I'm curious though, are there situations where there's a downside to using a `form`, or where a `form` cannot be used? The fact that a field cannot (at least potentially) be wrapped in a `form` may be a signal that it shouldn't be using validation!